### PR TITLE
Download example

### LINKED ISSUE #611: Add "download dataset" sample
I think a simple sample for the docs that shows how to download a bounding box of a dataset would be very helpful.

### PR DESCRIPTION
**Pull request by leowe**
### Description:
- Adds a download dataset example

### Issues:
- fixes #611

### CODE REVIEW

File: docs/src/webknossos-py/examples/download_image_data.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Upload Image Data
```
Comment (jstriebel) [inline docs/src/webknossos-py/examples/download_image_data.md:1]:
```suggestion
# Download Image Data
```

File: docs/src/webknossos-py/examples/download_image_data.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+Downloads part of a [webknossos dataset](https://webknossos.org/datasets/scalable_minds/l4dense_motta_et_al_demo/view?utm_source=site#2867,4361,1770,0,1.21) from a local webknossos instance using [`webknossos_context`](../../api/webknossos/client/context.md#webknossos_context).
```
Comment (jstriebel) [inline docs/src/webknossos-py/examples/download_image_data.md:3]: Should we set the `utm_source=site` in the docs now or is this copied from somewhere else? cc @normanrz 

```suggestion
Downloads part of a [webknossos dataset](https://webknossos.org/datasets/scalable_minds/l4dense_motta_et_al_demo/view) from a local webknossos instance using [`webknossos_context`](../../api/webknossos/client/context.md#webknossos_context).
```

File: docs/src/webknossos-py/index.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ To get started, check out the [installation instructions](installation.md) and t
 * [Intro to Dataset Usage](examples/dataset_usage.md)
 * [How to Upload a Dataset](examples/upload_image_data.md)<br/>
   ![Cell Dataset uploaded to webKnossos](examples/upload_image_data_dataset.jpg){: style="max-width: calc(100%/2 - 1em);"}
+* [How to Download a Dataset](examples/download_image_data.md)<br/>
```
Comment (jstriebel) [inline docs/src/webknossos-py/index.md:10]: The `<br/>` tag was just used for the images.
```suggestion
* [How to Download a Dataset](examples/download_image_data.md)
```

File: webknossos/webknossos/dataset/dataset.py [hunk 4]
```diff
@@ -227,6 +227,18 @@ def download(
...
+        """Downloads a dataset and returns the Dataset instance.
+
+        The `dataset_name` and `organization_name` specify which dataset to download.
+        The `bbox`, `layers`, and `mags` parameters specify which parts of the dataset to download.
+        If nothing is specified the whole image, all layers, and all mags are downloaded respectively.
+
+        The `path` and `exist_ok` parameter specify where to save the downloaded dataset and whether to overwrite
+        if the `path` exists.
+
+        The `webknossos_url` specifies in which webknossos instance to search for the dataset.
```
Comment (jstriebel) [inline webknossos/webknossos/dataset/dataset.py:239]:
```suggestion
        The `webknossos_url` specifies in which webknossos instance to search for the dataset. By default the
        configured url from `webknossos_context` is used, using https://webknossos.org as a fallback.
```

**Review by jstriebel — CHANGES_REQUESTED** [review]
Please also add this example to `test_examples.py` (see the existing ones for examples).

File: docs/src/webknossos-py/examples/download_image_data.md (hunk 2 above)
Comment (normanrz) [reply docs/src/webknossos-py/examples/download_image_data.md:3]: No, utm_source doesn't need to be part of links in the docs.

File: webknossos/tests/test_examples.py [hunk 5]
```diff
@@ -101,6 +101,19 @@ def test_upload_image_data() -> None:
         assert url.startswith("http://localhost:9000/datasets/Organization_X/cell_")
 
 
+@pytest.mark.block_network(allowed_hosts=[".*"])
+@pytest.mark.vcr(ignore_hosts=["gitlab.com"])
```
Comment (jstriebel) [inline webknossos/tests/test_examples.py:105]: I don't think that those should be added here.

File: docs/src/webknossos-py/examples/download_image_data.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Download Image Data
+
+Downloads part of a [webknossos dataset](https://webknossos.org/datasets/scalable_minds/l4dense_motta_et_al_demo/view) from a local webknossos instance using [`webknossos_context`](../../api/webknossos/client/context.md#webknossos_context).
```
Comment (jstriebel) [inline docs/src/webknossos-py/examples/download_image_data.md:3]:
```suggestion
Downloads part of a [webknossos dataset](https://webknossos.org/datasets/scalable_minds/l4dense_motta_et_al_demo/view) from [webknossos.org](https://webknossos.org) using [`webknossos_context`](../../api/webknossos/client/context.md#webknossos_context).
```

**Review by jstriebel — CHANGES_REQUESTED** [review]
Nice, thanks :+1: I think you still need to add the cassettes, and run `./test.sh --refresh-snapshots` therefore.

**Review by jstriebel — APPROVED** [review]
Happy when CI is :+1: (you still need the snapshots for the tests to succeed)

**Pull request merged**